sktime/forecasting/chronos.py

```diff
@@ -0,0 +1,268 @@
+
+        index_names = _y.index.names
+        if isinstance(_y.index, pd.MultiIndex):
+            _y = _frame2numpy(_y)
+        else:
+            _y = _y.values.reshape(1, -1, 1)
+
+        results = []
+        for i in range(_y.shape[0]):
+            _y_i = _y[i, :, 0]
+            _y_i = _y_i[-self.model_pipeline.model.config.context_length :]
+            prediction_results = self.model_pipeline.predict(
+                torch.Tensor(_y_i),
+                prediction_length,
+                num_samples=self._config["num_samples"],
+                temperature=self._config["temperature"],
+                top_k=self._config["top_k"],
+                top_p=self._config["top_p"],
+                limit_prediction_length=False,
+            )
+
+            values = np.median(prediction_results[0].numpy(), axis=0)
+            results.append(values)
+
+        pred = np.stack(results, axis=1)
+        if isinstance(_y_df.index, pd.MultiIndex):
+            ins = np.array(
+                list(np.unique(_y_df.index.droplevel(-1)).repeat(pred.shape[0]))
+            )
+            ins = [ins[..., i] for i in range(ins.shape[-1])] if ins.ndim > 1 else [ins]
+
+            idx = (
+                ForecastingHorizon(range(1, pred.shape[0] + 1), freq=self.fh.freq)
+                .to_absolute(self._cutoff)
+                ._values.tolist()
+                * pred.shape[1]
+            )
+            index = pd.MultiIndex.from_arrays(
+                ins + [idx],
+                names=_y_df.index.names,
+            )
+        else:
+            index = (
+                ForecastingHorizon(range(1, pred.shape[0] + 1))
+                .to_absolute(self._cutoff)
+                ._values
+            )
+        pred_out = fh.get_expected_pred_idx(_y, cutoff=self.cutoff)
+
+        pred = pd.DataFrame(
+            pred.reshape(-1, 1),
+            index=index,
+            columns=_y_df.columns,
+        )
+        dateindex = pred.index.get_level_values(-1).map(lambda x: x in pred_out)
+        pred.index.names = index_names
+
+        return pred.loc[dateindex]
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return ``"default"`` set.
+
+        Returns
+        -------
+        params : dict or list of dict
+        """
+        test_params = []
+        test_params.append(
+            {
+                "model_path": "amazon/chronos-t5-tiny",
+            }
+        )
+        test_params.append(
+            {
+                "model_path": "amazon/chronos-t5-tiny",
+                "config": {
+                    "num_samples": 20,
+                },
+                "seed": 42,
+            }
+        )
```

Comment:
I think the 2 cases are very identical

In the first case, by default the `num_samples` is None which still treats it the same way as with `num_samples=20` and returns output with shape of 20 at axis 1. So it doest not make any difference. I might be wrong here, despite of that, putting `num_samples` in the config does not really add any more coverage from what I see.

Reply:
sktime's tests require 2 test params per estimator, and there is no other test case that would add any more coverage, I would suggest we stick with these 2 test params untill we find a need for another one as we progress with adding more features like fine-tuning.